File: python-legacy-build/magic/magic_crit.py
from magic_settings import SET_INS

class Crit:
    def computeFCritChance(self, channeler_stacks, gconc):
        fcrit = round((0.05 * SET_INS.kalgSpec) + (0.01 * SET_INS.kalg) + (0.05 * SET_INS.reavers) + (0.12 * SET_INS.grim) + SET_INS.biting + (0.05 * gconc) + (0.04 * channeler_stacks),4)
        
        return fcrit
    
    def computeFCritDmg(self, fixed, variable, channeler_stacks):
        fcritAvg = (fixed + int(0.5 * (int(variable * 0.95) + variable)))
        
        if SET_INS.enchMeta == 1 and SET_INS.ring == 'Channelers':
            fcritAvg = fcritAvg + int(fcritAvg * 0.025 * channeler_stacks)
        else:
            pass
        
        return fcritAvg
    
    def computeNCritChance(self, fixed, variable, fcrit):
        #this is not technically accurate but making it accurate would require a pretty significant refator of the code which I don't want to do for something that is being replaced in a few weeks. It is close enough for now.
        maxHit = fixed + variable
        minCrit = int(maxHit * 0.95)
        ncrit = (1 - fcrit) * ((maxHit - minCrit) / variable)
        
        return ncrit
    
    def computeNCritDmg(self, fixed, variable, channeler_stacks):
        maxHit = fixed + variable
        minCrit = int(0.95 * maxHit)
        ncritAvg = int((maxHit + minCrit) / 2)
        
        if SET_INS.enchMeta == 1 and SET_INS.ring == 'Channelers':
            ncritAvg = ncritAvg + int(ncritAvg * 0.025 * channeler_stacks)
        else:
            pass
        
        return ncritAvg
    
    

File: magic/magic_crit.js
class Crit {
    calcFCritChance(channeler_stacks, gconc, kalg, kalgSpec, reavers, grim, biting) {
        let fcrit = parseFloat(
            (0.05 * kalgSpec) +
            (0.01 * kalg) +
            (0.05 * reavers) +
            (0.12 * grim) +
            biting +
            (0.05 * gconc) +
            (0.04 * channeler_stacks)
        ).toFixed(4);
        
        return fcrit;
    }
    
    calcFCritDmg(fixed, variable, channeler_stacks, ench) {
        let fcritAvg = fixed + Math.floor(0.5 * (Math.floor(variable * 0.95) + variable));
        
        if (ench === true) {
            fcritAvg = fcritAvg + Math.floor(fcritAvg * 0.025 * channeler_stacks);
        }

        return fcritAvg;
    }
    
    calcNCritChance(fixed, variable, fcrit) {
        const maxHit = fixed + variable;
        const minCrit = Math.floor(maxHit * 0.95);
        const ncrit = (1 - fcrit) * ((maxHit - minCrit) / variable);
        
        return ncrit;
    }
    
    calcNCritDmg(fixed, variable, channeler_stacks, ench) {
        const maxHit = fixed + variable;
        const minCrit = Math.floor(0.95 * maxHit);
        let ncritAvg = Math.floor((maxHit + minCrit) / 2);
        
        if (ench === true) {
            ncritAvg = ncritAvg + Math.floor(ncritAvg * 0.025 * channeler_stacks);
        }
        
        return ncritAvg;
    }
}

module.exports = Crit;
